packages/plugins/src/components/plugin-error-boundary/index.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+interface PluginErrorBoundaryProps {
+	/**
+	 * The name of the plugin that may encounter an errors.
+	 */
+	name: string;
+	/**
+	 * The child components to render.
+	 */
+	children: ReactNode;
+	/**
+	 * Callback function called when an error occurs.
+	 */
+	onError?: ( name: string, error: Error ) => void;
+}
+
+interface PluginErrorBoundaryState {
+	hasError: boolean;
+}
```

Comment:
If we,
- Use `React.ReactNode` without importing from `'react'`
- Move these two interfaces to a `types.ts` file in the same directory, and then use
```ts
import type {
	PluginErrorBoundaryProps as Props,
	PluginErrorBoundaryState as State,
} from './types';

export class PluginErrorBoundary extends Component< Props, State > {
	//
}
```

That way, we can preserve the git history of the file.